benchmark/nixlbench/src/utils/utils.cpp

```diff
@@ -205,18 +263,104 @@ uint64_t xferBenchConfig::gusli_bdev_byte_offset = 0;
-    benchmark_group = FLAGS_benchmark_group;
-    runtime_type = FLAGS_runtime_type;
-    worker_type = FLAGS_worker_type;
+xferBenchConfig::parseConfig(int argc, char *argv[]) {
+    cxxopts::Options options("nixlbench", "NIXL Benchmark Tool");
+
+    options.add_options()("help", "Print usage");
+
+    for (size_t i = 0; i < sizeof(xbench_params) / sizeof(xbench_params[0]); i++) {
+        const xferBenchParamInfo *param = &xbench_params[i];
+
+        switch (param->type) {
+        case XBPT_STRING:
+            options.add_options()(
+                param->name,
+                param->help,
+                cxxopts::value<std::string>()->default_value(param->def_value.str));
+            break;
+        case XBPT_BOOL:
+            options.add_options()(
+                param->name,
+                param->help,
+                cxxopts::value<bool>()->default_value(param->def_value.b ? "true" : "false"));
+            break;
+        case XBPT_UINT64:
+            options.add_options()(
+                param->name,
+                param->help,
+                cxxopts::value<uint64_t>()->default_value(std::to_string(param->def_value.u64)));
+            break;
+        case XBPT_INT32:
+            options.add_options()(
+                param->name,
+                param->help,
+                cxxopts::value<int32_t>()->default_value(std::to_string(param->def_value.i32)));
+            break;
+        default:
+            std::cerr << param->name << ": unsupported param type: " << param->type << std::endl;
+            return -1;
+        }
+    }
+
+    auto result = options.parse(argc, argv);
+    if (result.count("help")) {
+        std::cout << options.help() << std::endl;
+        return -1;
+    }
+
+    return loadParams(result);
+}
+
+// getParamValue() provides a parameter value, giving priority to explicitly passed
+// parameters over those specified in the config_file or set as defaults.
+template<class T>
+T
+xferBenchConfig::getParamValue(const std::unique_ptr<inih::INIReader> &ini,
+                               cxxopts::ParseResult &result,
+                               const char *name) {
+    if (ini && !result.count(name)) {
+        // config_file exists and the parameter is not specified explicitly ->
+        // try to read the value from config_file first
+        try {
+            return ini->Get<T>("global", name);
+        }
+        catch (const std::runtime_error &) {
+            // the parameter is not in the config_file -> fallback to ParseResult
+        }
+    }
+
+    return result[name].as<T>();
+}
+
+int
+xferBenchConfig::loadParams(cxxopts::ParseResult &result) {
+    std::unique_ptr<inih::INIReader> ini;
+
+    if (result.count("config_file")) {
+        /* if config_file parameter specified - try to read the config from file */
+        config_file = result["config_file"].as<std::string>();
+        try {
+            ini = std::make_unique<inih::INIReader>(config_file);
```

Comment:
Can we also add validation for config file here. Like checking for a global section